Drug Label Extraction Using Deep Learning (#3548)
section-engineering-education/engineering-education: content/articles/drug-labels-extraction-using-deep-learning/index.md

img = cv2.imread(img_path) 
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
```

The command `cv2.COLOR_BGR2RGB` reorders the color channels. By default, OpenCV imports images as BGR.

The final step involves visualizing our images and detections. 

Using matplotlib, let's first resize our display area.

```python
plt.figure(figsize=(15,15))
```
We then use the `draw_ocr` method to draw annotations on the image

```python
annotated = draw_ocr(img, boxes, texts, scores, font_path=font_path) 
```

Finally, we use matplotlib's `imshow()` method to visualize the image

```python
plt.imshow(annotated)
```
Let's take a look at the result. This's the same image we had earlier. We can see that it has extracted the text and drawn bounding boxes around it.

![Output image using PaddleOCR](/engineering-education/drug-labels-extraction-using-deep-learning/output.png)

Please find the full code to my code implementation [here](https://colab.research.google.com/drive/11EfSPacknXm16wWIkbyCCbyFSqVou0Ht).

### Wrapping up
One key feature of the PaddleOCR model is that it's extremely fast. We also note that with just a few lines of code, you can extract labels from prescribed medicine. Amazing, right? 

> In a nutshell, this tutorial shows you how you can use OCR to extract text from drug labels. The sizes and quality of your image will greatly impact your results. Try playing around with the code and see what you find.  

Reviewer: Alert readers that the colab version may give an error since the image file may not be available to them remotely. Or advise them to download them then use their image paths for the task. Or advise them to carry out the task locally for the best outcome ie. faster processing due to local CPU/GPU usage, ease of adding images, etc.

This technology is what our tutorial will be based on. We'll use OCR to enable us automatically extract drug labels/text from prescription medicine. 

We'll specifically use the PaddleOCR, previously referenced in a paper known as [PP-OCR: A Practical Ultra Lightweight OCR System](https://arxiv.org/pdf/2009.09941v3.pdf). This is a model that was originally built by the engineers at [Baidu](https://www.baidu.com/). 

### Prerequisites
To understand this tutorial, you need:
- To have either Google Colab or Jupyter Notebook installed.
- You need to be familiar with machine learning.
- You need to have Python and PaddleOCR installed.

### Table of contents
- [Installing PaddleOCR for Python](#installing-paddleocr-for-python)
- [Applying PaddleOCR model to images](#applying-paddleocr-model-to-images)
- [Extracting prescription medication labels using PaddleOCR](#extracting-prescription-medication-labels-using-paddleocr)
- [Wrapping up](#wrapping-up)

To go through with our drug label extraction, we need to perform three key things:

- Install and import our dependencies.
- Instantiate the model and detect.
- Visualize the results.

### Installing PaddleOCR for Python

> First, what we need to do is to open either your colab or notebook and install the dependencies. The dependency we want to install is the PaddleOCR. Please head on to Paddle OCR's Github [page](https://github.com/PaddlePaddle/PaddleOCR). 

Reviewer 2: We begin with opening a Jupyter notebook either on Colab or locally and install the required libraries.

> On the `Tutorial` section down below on the page, go to the `installation` link and click on it. On this page, you will be provided with a quick installation guide for the PaddleOCR.

Reviewer 2: Under the `Tutorial` section, navigate to the `installation` sub-section. A thorough guide to install PaddleOCR is available here (possibly attach the link instead of asking the reader to go to another page and find the link)

There are two parts to the installation process. First, you have to install `paddlepaddle` which is the underlying framework behind the PaddleOCR model. Secondly, we'll go ahead and install the `paddleOCR`.

The PaddleOCR model can run on both a GPU and CPU. GPUs are often preferred because of their greater memory bandwidth. 

If you have a GPU-enabled machine. use the following command to install the PaddleOCR:

> ```python

Reviewer 2: These would be bash commands and not Python commands

---
layout: engineering-education
status: publish
published: true
url: /drug-labels-extraction-using-deep-learning/
title: Drug Label Extraction Using Deep Learning
description: This tutorial will show the reader how to extract drug labels from prescription medicine using OCR.
author: wilkister-mumbi
date: 2021-09-13T00:00:00-18:30
topics: [Machine Learning]
excerpt_separator: <!--more-->
images:

  - url: /engineering-education/drug-labels-extraction-using-deep-learning/hero.png
    alt: Drug Label Image
---

> Optical Character Recognition (OCR) is a technology that can use optics to extract readable text into machine-encoded text. A large number of companies that process paper-based forms use OCR to extract texts from documents.

Reviewer: ```suggestion
Optical Character Recognition (OCR) uses optics to extract readable text into machine-encoded text. A large number of companies that process paper-based forms use OCR to extract texts from documents.
```

<!--more-->
Applying cutting-edge technologies to modern problems has enabled various problem-solving in healthcare, thereby improving people's lives. Machine learning has been used in hospitals to identify patients faster in disease identification and diagnosis, robotic surgery, and medical imaging diagnosis.

> One notable area of concern in healthcare that hasn't been explored a lot is drug label extraction. Optical Character Recognition (OCR) is a technology that can use optics to extract readable text into machine-encoded text.

Reviewer: ```suggestion
One notable area of concern in healthcare that hasn't been explored a lot is drug label extraction. Optical Character Recognition (OCR) uses optics to extract readable text into machine-encoded text.
```

> The OCR technology has been used on Google's visual translation service. Also, a large number of companies that process paper-based forms use OCR to extract texts from documents. Recently, most companies offering PDF services, such as Adobe, provided this feature on their app.   

Reviewer: ```suggestion
OCR has been used on Google's visual translation service. A large number of companies that process paper-based forms also use OCR to extract texts from documents. Recently, most companies offering PDF services, such as Adobe, provided this feature on their app.
```

> This technology is what our tutorial will be based on. We'll use OCR to enable us to extract drug labels/text from prescription medicine automatically. 

Reviewer: ```suggestion
We'll use OCR to enable us to automatically extract drug labels/text from prescription medicine.
```

There are two parts to the installation process. First, you have to install `paddlepaddle`, which is the underlying framework behind the PaddleOCR model. Secondly, we'll go ahead and install the `paddleOCR`.

The PaddleOCR model can run on both a GPU and CPU. GPUs are often preferred because of their greater memory bandwidth. 

If you have a GPU-enabled machine, use the following command to install the PaddleOCR:

```bash
!python3 -m pip install paddlepaddle-gpu==2.0.0 -i https://mirror.baidu.com/pypi/simple
```

If you have a CPU-enabled machine, use the following command to install the PaddleOCR:

```bash
!python3 -m pip install paddlepaddle==2.0.0 -i https://mirror.baidu.com/pypi/simple
```
> If you happen to get an error when installing, it's because you haven't installed the `pip` command. To install it, use the command, `pip3 install --upgrade pip`. Also, since we're installing it on our notebook, we have to include the exclamation mark `!` before the installation command.

> Let's now install PaddleOCR. It is a quick pip install. Let's type in the following command:

Reviewer: ```suggestion
Let's install PaddleOCR through a quick pip install, using the following command:
```
